Library/Homebrew/cmd/install.rb

```diff
@@ -139,6 +139,11 @@ def install
       perform_preinstall_checks
 
       formulae.each { |f| install_formula(f) }
+    rescue FormulaClassUnavailableError => e
+      # Need to re-raise before `FormulaUnavailableError` (superclass of this)
+      # is handled, as searching for a formula doesn't make sense here (the
+      # formula was found, but there's a problem with its implementation).
+      raise e
```

Comment:
I think this should be `ofail e.meesage` as you should not raise and interrupt another formula's installation. Noted there is a loop here.